### PR TITLE
DWR-1220 Add filters to the API

### PR DESCRIPTION
**Pull request by agorina**

### CODE REVIEW

File: aggregate-service/src/main/java/org/opentestsystem/rdw/olap/repository/impl/JdbcCustomAggregateReportRowRepository.java [hunk 1]
```diff
@@ -66,8 +66,8 @@ private static void validate(final ReportQuery query) {
         if (!query.isIncludeAllDistricts() && !query.getDistrictIds().isEmpty()) parameters.addValue("district_ids", query.getDistrictIds());
         if (!query.getDistrictIdsWithAllSchools().isEmpty()) parameters.addValue("school_district_ids", query.getDistrictIdsWithAllSchools());
         if (!query.getSchoolIds().isEmpty()) parameters.addValue("school_ids", query.getSchoolIds());
-        if (!query.getCompletenessCodes().isEmpty()) parameters.addValue("completeness_codes", query.getCompletenessCodes());
-        if (!query.getAdministrativeConditionCodes().isEmpty()) parameters.addValue("administration_condition_codes", query.getAdministrativeConditionCodes());
+        if (!query.getCompleteness().isEmpty()) parameters.addValue("completeness_codes", query.getCompleteness());
```
Comment (agorina) [inline aggregate-service/src/main/java/org/opentestsystem/rdw/olap/repository/impl/JdbcCustomAggregateReportRowRepository.java]: Just renamed.

File: aggregate-service/src/main/java/org/opentestsystem/rdw/olap/web/ReportQueryArgumentResolver.java [hunk 2]
```diff
@@ -50,6 +50,9 @@ public Object resolveArgument(final MethodParameter parameter,
 
         final String dimensions = webRequest.getParameter("dimensionTypes");
         final String districtsRaw = webRequest.getParameter("districts");
+        final String genders = webRequest.getParameter("genders");
+        final String completeness = webRequest.getParameter("completeness");
+        final String administrativeConditions = webRequest.getParameter("administrativeConditions");
```
Comment (agorina) [inline aggregate-service/src/main/java/org/opentestsystem/rdw/olap/web/ReportQueryArgumentResolver.java]: This is the real change.

File: aggregate-service/src/main/java/org/opentestsystem/rdw/olap/sqlbuilder/CustomAggregateQueryProvider.java [hunk 3]
```diff
@@ -102,8 +102,8 @@ private SelectQuery buildSql(final Collection<String> addOns, final DimensionTyp
     private String buildCustomAggregateSqlString(final ReportQuery query, final ImmutableSet<String> addOns, final DimensionType dimensionType) {
         final Set<String> addOnsWithFilters = new HashSet(addOns);
 
-        if (!query.getAdministrativeConditionCodes().isEmpty()) addOnsWithFilters.add("administration_condition");
-        if (!query.getCompletenessCodes().isEmpty()) addOnsWithFilters.add("completeness");
+        if (!query.getAdministrativeConditions().isEmpty()) addOnsWithFilters.add("administration_condition");
+        if (!query.getCompleteness().isEmpty()) addOnsWithFilters.add("completeness");
```
Comment (malaffoon) [inline aggregate-service/src/main/java/org/opentestsystem/rdw/olap/sqlbuilder/CustomAggregateQueryProvider.java]: You made the order different (than the previous file) just to mess with my OCD, didn't you?

File: aggregate-service/src/main/java/org/opentestsystem/rdw/olap/repository/impl/JdbcCustomAggregateReportRowRepository.java [hunk 4]
```diff
@@ -66,8 +66,8 @@ private static void validate(final ReportQuery query) {
...
+        if (!query.getAdministrativeConditions().isEmpty()) parameters.addValue("administration_condition_codes", query.getAdministrativeConditions());
```
Comment (malaffoon) [inline aggregate-service/src/main/java/org/opentestsystem/rdw/olap/repository/impl/JdbcCustomAggregateReportRowRepository.java]: Should the param name change?

File: aggregate-service/src/main/java/org/opentestsystem/rdw/olap/sqlbuilder/CustomAggregateQueryProvider.java (hunk 3 above)
Comment (agorina) [reply aggregate-service/src/main/java/org/opentestsystem/rdw/olap/sqlbuilder/CustomAggregateQueryProvider.java]: Yes, that was exactly my goal :)

File: aggregate-service/src/main/java/org/opentestsystem/rdw/olap/repository/impl/JdbcCustomAggregateReportRowRepository.java (hunk 4 above)
Comment (agorina) [reply aggregate-service/src/main/java/org/opentestsystem/rdw/olap/repository/impl/JdbcCustomAggregateReportRowRepository.java]: Yes, will change.

**Pull request merged**